src/test/java/com/pivovarit/collectors/parallelToCollection/ToCollectionExceptionPropagationTest.java

```diff
@@ -2,66 +2,98 @@
+        ).flatMap(identity());
+    }
 
-        assertThatThrownBy(IntStream.range(0, 10).boxed()
-          .map(i -> supplier(() -> throwing(i)))
-          .collect(parallelToCollection(ArrayList::new, executor, 10))::join)
-          .isInstanceOf(CompletionException.class)
-          .hasCauseExactlyInstanceOf(IllegalArgumentException.class);
+    private static Stream<DynamicTest> forCollector(TriFunction<Function<Integer, Integer>, Executor, Integer, Collector<Integer, List<CompletableFuture<Integer>>, ? extends CompletableFuture<? extends Collection<Integer>>>> collector, String name) {
+        return of(
+                shouldCollectWithMappingAndNotSwallowException(collector, name),
+                shouldCollectWithMappingAndNotSwallowExceptionUnbounded(collector, name));
     }
 
-    @Test
-    void shouldCollectToCollectionAndNotSwallowExceptionUnbounded() {
-        // given
-        executor = threadPoolExecutor(10);
+    private static Stream<DynamicTest> forCollector(BiFunction<Executor, Integer, Collector<Supplier<Integer>, List<CompletableFuture<Integer>>, ? extends CompletableFuture<? extends Collection<Integer>>>> collector, String name) {
+        return of(
+                shouldCollectAndNotSwallowException(collector, name),
+                shouldCollectAndNotSwallowExceptionUnbounded(collector, name));
+    }
 
-        assertThatThrownBy(IntStream.range(0, 10).boxed()
-          .map(i -> supplier(() -> throwing(i)))
-          .collect(parallelToCollection(ArrayList::new, executor))::join)
-          .isInstanceOf(CompletionException.class)
-          .hasCauseExactlyInstanceOf(IllegalArgumentException.class);
+    private static DynamicTest shouldCollectWithMappingAndNotSwallowException(TriFunction<Function<Integer, Integer>, Executor, Integer, Collector<Integer, List<CompletableFuture<Integer>>, ? extends CompletableFuture<? extends Collection<Integer>>>> collector, String name) {
+        return DynamicTest.dynamicTest(format("%s: should collect with mapping and not swallow exception", name),
+                () -> assertThatThrownBy(IntStream.range(0, 10).boxed()
+                        .collect(collector.apply(TestUtils::throwing, threadPoolExecutor(10), 10))::join)
+                        .isInstanceOf(CompletionException.class)
+                        .hasCauseExactlyInstanceOf(IllegalArgumentException.class));
     }
 
+    private static DynamicTest shouldCollectAndNotSwallowException(BiFunction<Executor, Integer, Collector<Supplier<Integer>, List<CompletableFuture<Integer>>, ? extends CompletableFuture<? extends Collection<Integer>>>> collector, String name) {
+        return DynamicTest.dynamicTest(format("%s: should collect and not swallow exception", name),
+                () -> assertThatThrownBy(IntStream.range(0, 10).boxed()
+                        .map(i -> supplier(() -> throwing(i)))
+                        .collect(collector.apply(threadPoolExecutor(10), 10))::join)
```

Comment:
`threadPoolExecutor(10), 10)` need to be closed after usage to enable reclamation of resources - we'll not end up with an OutOfMemoryError